src/manager.rs

```diff
@@ -165,6 +180,10 @@ impl Manager {
             Box::pin(async move {
                 shutdown_signal.cancel();
                 consumer_join_handle.await.expect("Failed to stop consumer");
+                
+                results_worker.await.expect("Failed to stop vector worker");
+                results_shutdown_sender.send(()).expect("Failed to send shutdown signal");
```

Comment:
not sure if this line is necessary, i'm not sure if we have to send a shutdown signal here as we don't for our other join handles?

Reply:
We send the signal right above here, line 181